<a href="https://colab.research.google.com/github/notgian/csmodel-gr4-backup-repo/blob/main/CSMODEL_GRP4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Dataset Description

## Some Necessary Imports
We first import the necessary packages and the dataset itself. For portability purposes, the dataset will be downloaded via kagglehub.

In [5]:
import pandas as pd
import numpy as np

In [12]:
# Importing the dataset via kagglehub
import kagglehub
import os

csv_path = kagglehub.dataset_download("fatemehmehrparvar/obesity-levels")
csv_file = os.listdir(csv_path)[0] # The only file downloaded is the obesity csv
csv_filepath = os.path.join(csv_path, csv_file)

obesity_df = pd.read_csv(csv_filepath)

Using Colab cache for faster access to the 'obesity-levels' dataset.


## Describing the Dataset
We then view the different properties of the dataframe to find out more about the data.

In [17]:
obesity_df.shape

(2111, 17)

Given the shape of the dataframe, the dataset contains 2,111 rows and 17 columns. Next, we view the head of the data to view the different columns.

In [18]:
obesity_df.head()

Age  Gender  Height  Weight        CALC FAVC  FCVC  NCP  SCC SMOKE  CH2O  \
0  21.0  Female    1.62    64.0          no   no   2.0  3.0   no    no   2.0   
1  21.0  Female    1.52    56.0   Sometimes   no   3.0  3.0  yes   yes   3.0   
2  23.0    Male    1.80    77.0  Frequently   no   2.0  3.0   no    no   2.0   
3  27.0    Male    1.80    87.0  Frequently   no   3.0  3.0   no    no   2.0   
4  22.0    Male    1.78    89.8   Sometimes   no   2.0  1.0   no    no   2.0   

  family_history_with_overweight  FAF  TUE       CAEC                 MTRANS  \
0                            yes  0.0  1.0  Sometimes  Public_Transportation   
1                            yes  3.0  0.0  Sometimes  Public_Transportation   
2                            yes  2.0  1.0  Sometimes  Public_Transportation   
3                             no  2.0  0.0  Sometimes                Walking   
4                             no  0.0  0.0  Sometimes  Public_Transportation   

            NObeyesdad  
0        Normal_Weight  
1        Normal_Weight  
2        Normal_Weight  
3   Overweight_Level_I  
4  Overweight_Level_II

*observe the weird names for the columns. insert explanation here...*
*just describe each one at a time*

**1. Age**
  - The person's age.

**2. Gender**

**3. Height**

**4. Weight**

**5. CALC**

**6. FAVC**

**7. FCVC**

**8. NCP**

**9. SCC**

**10. SMOKE**

**11. CH2O**

**12. family_history_with_overweight**

**13. FAF**

**14. TUE**

**15. CAEC**

**16. MTRANS**

**17. NObeyesdad**


# Data Cleaning

# Exploratory Data Analysis

# Research Question
